### PR TITLE
Merchant dashboard User story 1 - 3

### LINKED ISSUE #2: 1. Merchant Dashboard
As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the name of my merchant

### LINKED ISSUE #3: 2. Merchant Dashboard Links
As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see link to my merchant items index (/merchants/:merchant_id/items)
And I see a link to my merchant invoices index (/merchants/:merchant_id/invoices)

### LINKED ISSUE #5: 3. Merchant Dashboard Statistics - Favorite Customers
3. Merchant Dashboard Statistics - Favorite Customers

As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the names of the top 5 customers
who have conducted the largest number of successful transactions with my merchant
And next to each customer name I see the number of successful transactions they have
conducted with my merchant

### PR DESCRIPTION
**Pull request by pcbennett108**
1. Merchant Dashboard
As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the name of my merchant

2. Merchant Dashboard Links
As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see link to my merchant items index (/merchants/:merchant_id/items)
And I see a link to my merchant invoices index (/merchants/:merchant_id/invoices)

3. Merchant Dashboard Statistics - Favorite Customers
As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the names of the top 5 customers
who have conducted the largest number of successful transactions with my merchant
And next to each customer name I see the number of successful transactions they have
conducted with my merchant

Added validations
Added relationships
Added MVCs for user stories
Added methods for Merchant and Transactions
Added RSpec test data to Merchant and dashboard

#2 #3 #5

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -1,3 +1,19 @@
 class Merchant < ApplicationRecord
+  validates :name, presence: true
+
   has_many :items
+  has_many :invoices, through: :items
```
Comment (brad2412) [inline app/models/merchant.rb:5]: Thanks for adding the associations

**Review by brad2412 — APPROVED** [review]

File: app/models/customer.rb [hunk 2]
```diff
@@ -1,18 +1,21 @@
 class Customer < ApplicationRecord
   has_many :invoices
   has_many :transactions, through: :invoices
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
```
Comment (algibson1) [inline app/models/customer.rb:6]: These can be chained together like `validates_presence_of :first_name, :last_name`

**Review by topher-nullset — APPROVED** [review]
looks good to me!

File: app/models/customer.rb (hunk 2 above)
Comment (brad2412) [reply app/models/customer.rb:6]: Oh good idea!

File: app/models/transaction.rb [hunk 3]
```diff
@@ -1,3 +1,9 @@
 class Transaction < ApplicationRecord
   belongs_to :invoice
+
+  validates :result, presence: true
+  validates :credit_card_number, presence: true
+  validates :credit_card_expiration_date, presence: true
+
+  enum :result, { failed: 0, success: 1 }
```
Comment (algibson1) [inline app/models/transaction.rb:8]: I know this table wasn't included in our instructions for enums so mainly I want to keep in mind the question for instructors: Are they useful when there are only two results or would a boolean be better?

**Pull request merged**